services/api/src/typeDefs.js

```diff
@@ -984,19 +1001,19 @@ const typeDefs = gql`
     patch: UpdateTaskPatchInput!
   }
 
-  input AddOpenshiftInput {
+  input AddKubernetesInput {
```

Comment:
It would be odd here to use AddKubernetesInput to add an apiType of OPENSHIFT, and maybe just continues the same misnaming?  Why do we not just AddDeployTarget (/Update/Delete the same)

Reply:
I don't think so, in the end OpenShift is of type Kubernetes, it's just a special type of Kubernetes. so I think this is correct, you add a kubernetes cluster and tell lagoon that this is an openshift.

While if we would use `AddDeployTarget` you also need a way to tell graphql that you want to add an object `Kubernetes` and also different deployTargets might have different attributes (like a mesosphere cluster might not have a remoteShellHost)

Reply:
I'm happy as long as we do have that flexibility - in the end Kubernetes is a type implementing deployTarget.  If each deployTarget type has different required fields, then this is definitely easier :+1: 

As you were